pyiron_atomistics/calphy/job.py

```diff
@@ -731,10 +849,97 @@ def collect_ediff(self):
+        """
+        f_ediff = []
+        b_ediff = []
+        f_vol = []
+        b_vol = []
+        f_press = []
+        b_press = []
+
+        for i in range(1, self.input.n_iterations + 1):
+            fwdfilename = os.path.join(self.working_directory, "%s.forward_%d.dat" % (mode, i))
+            bkdfilename = os.path.join(self.working_directory, "%s.backward_%d.dat" % (mode, i))
+
+            fdx, fp, fvol, flambda = np.loadtxt(fwdfilename, unpack=True, comments="#")
+            bdx, bp, bvol, blambda = np.loadtxt(bkdfilename, unpack=True, comments="#")
+
+            fdx /= flambda
+            bdx /= blambda
+
+            f_ediff.append(fdx)
+            b_ediff.append(bdx)
+
+            f_vol.append(fvol)
+            b_vol.append(bvol)
+
+            f_press.append(fp)
+            b_press.append(bp)
+
+        return f_ediff, b_ediff, f_vol, b_vol, f_press, b_press, flambda, blambda
+
+    def get_positions(self):
+        """
+        Collect positions and cells
+        """
+        fwd_positions = []
+        bkd_positions = []
+        fwd_cells = []
+        bkd_cells = []
+
+        for i in range(1, self.input.n_iterations + 1):
+            fwdfilename = os.path.join(self.working_directory, "traj.ts.forward_%d.dat" % (i))
+            bkdfilename = os.path.join(self.working_directory, "traj.ts.backward_%d.dat" % (i))
+            
+            fp = []
+            fc = []
+            bp = []
+            bc = []
+
+            if os.path.exists(fwdfilename):
+                traj = Trajectory(fwdfilename)
+                for x in traj.nblocks:
+                    aseobj = traj[x].to_ase(species=self.calc.element)
+                    fp.append(aseobj.positions)
+                    fc.append(list(aseobj.cell))
+            if os.path.exists(bkdfilename):
+                traj = Trajectory(bkdfilename)
+                for x in traj.nblocks:
+                    aseobj = traj[x].to_ase(species=self.calc.element)
+                    bp.append(aseobj.positions)
+                    bc.append(list(aseobj.cell))
+
+            fwd_positions.append(fp)
+            bkd_positions.append(bp)
+            fwd_cells.append(fc)
+            bkd_positions.append(bc)
+
+        return fwd_positions, bkd_positions, fwd_cells, bkd_cells
+
     def collect_output(self):
         self.collect_general_output()
         self.to_hdf()
 
+    def db_entry(self):
+        """
+        Generate the initial database entry
+        Returns:
+            (dict): db_dict
+        """
+        db_dict = super(Calphy, self).db_entry()
+        if self.structure:
+            if isinstance(self.structure, Atoms):
+                parent_structure = self.structure.get_parent_basis()
+            else:
+                parent_structure = self.structure.copy()
+            db_dict["ChemicalFormula"] = parent_structure.get_chemical_formula()
+
+        db_dict["projectpath"] = self.project_hdf5.project_path
```

Comment:
Is this not provided by `super()`?

Reply:
No, the `projectpath` remained empty.

Reply:
Sounds like a bug then.  Ping @liamhuber @niklassiemer didn't we have a similar situation somewhere the projectpath is not correctly set up depending on some configuration issue?

Reply:
# TL;DR
if `projectpath` remained empty, you probably are not in one of the defined `PROJECT_PATHS` of your settings and  `PROJECT_CHECK_ENABLED` is `False`.
The handling inside the `super()` call should be the right behavior!
```suggestion
```

# Full answer
We have the very confusing situation that our notion of 'projectpath' changes depending on where in the code we are...:

- In the database (and the database code) we have 'projectpath' being the paths defined in our settings and 'project' being the path to the project
- In the jobs we have 'projectpath' being the path to the project and 'root_path' being the corresponding path defined in the settings.

In the `db_entry` these two worlds are translated to each other:
```python
            "projectpath": self.project_hdf5.root_path,
            "project": self.project_hdf5.project_path,
```
is the definition in [`GenericJob.db_entry()`](https://github.com/pyiron/pyiron_base/blob/19d6c3c5180fee5b0c0ccf91f1ec516a200cc624/pyiron_base/jobs/job/generic.py#L1167-L1168)

Depending on the settings (`PROJECT_PATHS` and `PROJECT_CHECK_ENABLED` definitions) 
- we split the full path  `/settings/project/path/path/to/actual/project` into `/settings/project/path/` and `path/to/actual/project`,
- while we 'split' `/not/a/settings/project/path/path/to/actual/project` into `None` and `/not/a/settings/project/path/path/to/actual/project` if `PROJECT_CHECK_ENABLED=False`
- and `raise ValueError` else. 
Splitting of `root_path` is done in [database.manager.DatabaseManager.top_path](https://github.com/pyiron/pyiron_base/blob/19d6c3c5180fee5b0c0ccf91f1ec516a200cc624/pyiron_base/database/manager.py#L213-L235)